ftests/036-cgset-multi_thread.py

```diff
@@ -0,0 +1,138 @@
+#
+# Basic cgrules functionality test
+#
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Author: Kamalesh Babulal <kamalesh.babulal@oracle.com>
+#
+
+#
+# This library is free software; you can redistribute it and/or modify it
+# under the terms of version 2.1 of the GNU Lesser General Public License as
+# published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, see <http://www.gnu.org/licenses>.
+#
+
+from cgroup import Cgroup, CgroupVersion
+from process import Process
+from run import Run
+import consts
+import ftests
+import sys
+import os
+
+CONTROLLER='cpu'
+PARENT_CGNAME='036cgset'
+CHILD_CGNAME='childcg'
+
+SETTING = 'cgroup.type'
+AFTER = 'threaded'
+THREADS = 3
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if config.args.container:
+        result = consts.TEST_SKIPPED
+        cause = "This test cannot be run within a container"
+        return result, cause
+
+    if CgroupVersion.get_version(CONTROLLER) != CgroupVersion.CGROUP_V2:
+        result = consts.TEST_SKIPPED
+        cause = "This test requires the cgroup v2"
+        return result, cause
+
+    return result, cause
+
+def setup(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    child_cgpath = os.path.join(PARENT_CGNAME, CHILD_CGNAME)
+
+    Cgroup.create(config, CONTROLLER, PARENT_CGNAME)
+    Cgroup.create(config, CONTROLLER, child_cgpath)
+    
+    Cgroup.set(config, child_cgpath, SETTING, AFTER)
+
+    after = Cgroup.get(config, controller=None, cgname=child_cgpath,
+            setting=SETTING, print_headers=False, values_only=True)
+
+    if after != AFTER:
+        result = consts.TEST_FAILED
+        cause = "cgget expected {} but received {}".format(AFTER, after)
+        return result, cause
+
+    return result, cause
+
+def get_cgroup_threads(config, cgname):
+    threads = Cgroup.get(config, controller=None, cgname=cgname,
+            setting="cgroup.threads", print_headers=False,values_only=True)
+
+    threads = threads.replace("\n", "")
+    threads = threads.replace("\t", " ")
+    threads = threads.split(" ")
+
+    return threads
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    child_cgpath = os.path.join(PARENT_CGNAME, CHILD_CGNAME)
+
+    pid = config.process.create_threaded_process_in_cgroup(config,
+                    CONTROLLER, PARENT_CGNAME, THREADS)
+
+    threads = get_cgroup_threads(config, PARENT_CGNAME)
+    thread_tid = threads[1]
+
+    Cgroup.set(config, child_cgpath, "cgroup.threads", thread_tid)
+    
+    cg_thread_tid = Cgroup.get(config, controller=None, cgname=child_cgpath,
```

Comment:
Not your fault here, but many of the tests have this exact logic - get, compare, then fail if comparison fails.  I've added an enhancement request to streamline this.
https://github.com/libcgroup/libcgroup-tests/issues/31

Let's obviously save that work for a different pull request.

Reply:
I added a pull request to address this copy-pasta.  Let me know what you think.
https://github.com/libcgroup/libcgroup-tests/pull/32